src/cpp/dds/publisher/qos/PublisherQos.cpp

```diff
@@ -0,0 +1,205 @@
+    {
+        presentation = qos.presentation;
+        presentation.hasChanged = true;
+    }
+    if(qos.partition.getNames().size()>0)
+    {
+        partition = qos.partition;
+        partition.hasChanged = true;
+    }
+
+    if (topic_data.getValue() != qos.topic_data.getValue())
+    {
+        topic_data = qos.topic_data;
+        topic_data.hasChanged = true;
+    }
+    if (group_data.getValue() != qos.group_data.getValue())
+    {
+        group_data = qos.group_data;
+        group_data.hasChanged = true;
+    }
+    if(first_time || durability_service.history_kind != qos.durability_service.history_kind ||
+            durability_service.history_depth != qos.durability_service.history_depth ||
+            durability_service.max_instances != qos.durability_service.max_instances ||
+            durability_service.max_samples != qos.durability_service.max_samples||
+            durability_service.max_samples_per_instance != qos.durability_service.max_samples_per_instance ||
+            durability_service.service_cleanup_delay != qos.durability_service.service_cleanup_delay
+      )
+    {
+        durability_service = qos.durability_service;
+        durability_service.hasChanged = true;
+    }
+    if(lifespan.duration != qos.lifespan.duration )
+    {
+        lifespan = qos.lifespan;
+        lifespan.hasChanged = true;
+    }
+    if(qos.ownership_strength.value !=ownership_strength.value)
+    {
+        ownership_strength = qos.ownership_strength;
+        ownership_strength.hasChanged = true;
+    }
+    if (first_time)
+    {
+        disable_positive_acks = qos.disable_positive_acks;
+        disable_positive_acks.hasChanged = true;
+    }
+}
+
+bool PublisherQos::check_qos() const
+{
+    using namespace fastrtps;
+    if(durability.kind == PERSISTENT_DURABILITY_QOS)
+    {
+        logError(RTPS_QOS_CHECK,"PERSISTENT Durability not supported");
+        return false;
+    }
+    if(destination_order.kind == BY_SOURCE_TIMESTAMP_DESTINATIONORDER_QOS)
+    {
+        logError(RTPS_QOS_CHECK,"BY SOURCE TIMESTAMP DestinationOrder not supported");
+        return false;
+    }
+    if(reliability.kind == BEST_EFFORT_RELIABILITY_QOS && ownership.kind == EXCLUSIVE_OWNERSHIP_QOS)
+    {
+        logError(RTPS_QOS_CHECK,"BEST_EFFORT incompatible with EXCLUSIVE ownership");
+        return false;
+    }
+    if(liveliness.kind == AUTOMATIC_LIVELINESS_QOS || liveliness.kind == MANUAL_BY_PARTICIPANT_LIVELINESS_QOS)
+    {
+        if(liveliness.lease_duration < c_TimeInfinite && liveliness.lease_duration <= liveliness.announcement_period)
+        {
+            logError(RTPS_QOS_CHECK,"WRITERQOS: LeaseDuration <= announcement period.");
+            return false;
+        }
+    }
+    return true;
+}
+
+
+bool PublisherQos::can_qos_be_updated(const PublisherQos& qos) const
+{
+    using namespace fastrtps;
+    bool updatable = true;
+    if(	durability.kind != qos.durability.kind)
```

Comment:
```suggestion
    if (durability.kind != qos.durability.kind)
```

Reply:
I moved this file, but I did the change...